lib/ui/serverlessFunctionLogs.ts

```diff
@@ -82,20 +108,32 @@ function processLog(log, options) {
   }
 }
 
-function processLogs(logsResp, options) {
-  if (!logsResp || (logsResp.results && !logsResp.results.length)) {
-    return 'No logs found.';
-  } else if (logsResp.results && logsResp.results.length) {
-    return logsResp.results
-      .map(log => {
+function isLogsResponse(
+  logsResp: LogsResponse | Log
+): logsResp is LogsResponse {
+  return (
+    'results' in logsResp &&
+    Array.isArray(logsResp.results) &&
+    logsResp.results !== undefined &&
+    logsResp.results.length > 0
```

Comment:
I think these last two checks can be removed.  They don't really determine the type are and more logic checks.  It seems like a `LogsResponse` with result = [] is still a valid logs response.  `Array.isArray` should also handle `undefined` [example](https://jsbin.com/betihisifu/edit?js,console)